internal/pkg/describe/lb_web_service.go

```diff
@@ -289,36 +291,80 @@ type WebServiceRoute struct {
 	URL         string `json:"url"`
 }
 
-// ServiceDiscovery contains serialized service discovery info for an service.
-type ServiceDiscovery struct {
+// InternalEndpoint contains serialized internal endpoint (for both service connect and service discovery) info for an ECS service.
+type InternalEndpoint struct {
 	Environment []string `json:"environment"`
-	Namespace   string   `json:"namespace"`
+	Endpoint    string   `json:"endpoint"`
+}
+
+type internalEndpoints []*InternalEndpoint
+
+type serviceDiscoveries internalEndpoints
+
+func (sds *serviceDiscoveries) collectServiceDiscoveryEndpoints(descr envDescriber, svc, env, port string) error {
```

Comment:
Let me modify the method name and i think it'll make more sense.

Reply:
i'm still not clear on why we should have separate types for `serviceDiscoveries` and `serviceConnects`....

Reply:
Because we have multiple env so we actually keep adding things to `sds *serviceDiscoveries`. using `collectServiceDiscoveryEndpoints() ([]*InternalEndpoint, error)` will create extra space every time. if we use `collectServiceDiscoveryEndpoints() (*[]*InternalEndpoint, error)` its kinda weird.

Reply:
One alternative is `collectServiceDiscoveryEndpoints(sds *serviceDiscoveries, descr envDescriber, svc, env, port string) error` but its essentially the same.

Reply:
oh i see. it basically takes care of the `append` from extra environments, instead of
```
var sdEndpoints []internalEndpoints
endpoints := getSDEndpoints()
append(sdEndpoints, endpoints...)
```

that makes sense to me! thanks 😊